src/components/include/DefaultImportModalDialog.jsx

```diff
@@ -0,0 +1,152 @@
+import { Made } from "@exabyte-io/made.js";
+import { defaultMaterialConfig } from "@exabyte-io/made.js/lib/material";
+import { PropTypes } from "prop-types";
+import React from "react";
+import { ModalBody, ModalHeader } from "react-bootstrap";
+import NPMsAlert from "react-s-alert";
+import _ from "underscore";
+
+import { Material } from "../../material";
+import BasisText from "../source_editor/BasisText";
+import { ModalDialog } from "./ModalDialog";
+
+const mockLatticeConstantMultiplicator = 2.0;
+
+class DefaultImportModalDialog extends ModalDialog {
+    constructor(props) {
+        super(props);
+        this.state = {
+            xyz: "Paste your XYZ",
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleFileRead = this.handleFileRead.bind(this);
+
+        this.reader = new FileReader();
+        this.reader.onloadend = this.handleFileRead;
+    }
+
+    handleFileRead(evt) {
+        this.setState({ xyz: evt.target.result });
+    }
+
+    handleSubmit() {
+        if (!this.BasisTextComponent.state.isContentValidated) return;
+
+        const basisConfig = Made.parsers.xyz.toBasisConfig(
+            this.state.xyz,
+            Made.ATOMIC_COORD_UNITS.cartesian,
+        );
+
+        // determine the max distance to use for cell for user's XYZ
+        const x = [];
+        const y = [];
+        const z = [];
+        basisConfig.coordinates.forEach((item) => {
+            x.push(item.value[0]);
+            y.push(item.value[1]);
+            z.push(item.value[2]);
+        });
+
+        const diffX = _.max(x) - _.min(x);
+        const diffY = _.max(y) - _.min(y);
+        const diffZ = _.max(z) - _.min(z);
+        const mockLatticeConstant = _.max([diffX, diffY, diffZ]) * mockLatticeConstantMultiplicator;
+
+        const material = new Made.Material(defaultMaterialConfig);
+
+        const latticeConfig = {
+            ...material.lattice,
+            ...{
+                a: mockLatticeConstant,
+                b: mockLatticeConstant,
+                c: mockLatticeConstant,
+            },
+        };
+        const lattice = new Made.Lattice(latticeConfig);
+
+        const basis = new Made.Basis({
+            ...basisConfig,
+            cell: lattice.vectorArrays,
+        });
+
+        const newMaterialConfig = {
+            ...material.toJSON(),
+            ...{
+                basis: basis.toJSON(),
+                lattice: lattice.toJSON(),
+                name: `${this.reader.currentFilename} - ${basis.formula}`,
+            },
+        };
+        const newMaterial = new Material(newMaterialConfig);
+        newMaterial.cleanOnCopy();
+
+        Made.tools.material.getBasisConfigTranslatedToCenter(newMaterial);
+        this.props.onSubmit([newMaterial]);
```

Comment:
This function does the job, but it's a bit too long. Here's what we can do:

1. Isolate the Math-related portions to Made
2. Create one more default material config in Made to have a simple cubic lattice and import here
3. Keep only 5-10 lines here with clear naming - e.g. `determineMaxInteratomicDistance(material)` etc.